src/logic/map_objects/tribes/ship.h

```diff
@@ -319,6 +328,7 @@ struct Ship : Bob {
 		Serial ware_economy_serial_;
 		Serial worker_economy_serial_;
 		std::vector<std::pair<uint32_t, uint32_t>> destinations_;
+		int32_t capacity_ = 0U;
```

Comment:
Mismatch between signed and unsigned.

I suspect that it's only signed for savegame compatibility? In that case, you could use `std::numeric_limits<uintxx_t>::max()` as a flag rather than `-1`.